src/Control/Eff/Cut.hs

```diff
@@ -2,6 +2,8 @@
 {-# LANGUAGE PatternGuards #-}
 {-# LANGUAGE DeriveDataTypeable #-}
 {-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE ScopedTypeVariables #-}
```

Comment:
First error message (w/oTypeFamilies):

```src/Control/Eff/Cut.hs:70:2:
    Illegal equational constraint Data.OpenUnion.Internal.OpenUnion2.Member'
                                    Choose r
                                  ~ 'True
    (Use GADTs or TypeFamilies to permit this)
    When checking that `loop' has the inferred type
      loop :: forall r a.
              (Data.OpenUnion.Internal.OpenUnion2.Member' Choose r ~ 'True) =>
              [Eff (Exc CutFalse :> r) a]
              -> Free (Union (Exc CutFalse :> r)) a -> Eff r a
    In an equation for `call':
        call
          = loop []
          where
              loop jq
                = freeMap
                    (\ x -> return x `mplus'` next jq)
                    (\ u
                       -> case decomp u of {
                            Right (Exc CutFalse) -> ...
                            Left u' -> ... })
              check ::
                Member Choose r =>
                [Eff (Exc CutFalse :> r) a]
                -> Union r (Eff (Exc CutFalse :> r) a) -> Eff r a
              check jq u | Just (Choose [] _) <- prj u = next jq
              check jq u | Just (Choose [x] k) <- prj u = loop jq (k x)
              check jq u | Just (Choose lst k) <- prj u = next $ map k lst ++ jq
              check jq u = send u >>= loop jq
              next :: Member Choose r => [Eff (Exc CutFalse :> r) a] -> Eff r a
```

After adding type families one of errors:

```
src/Control/Eff/Cut.hs:82:46:
    Could not deduce (r ~ r1)
    from the context (Member Choose r)
      bound by the type signature for
                 call :: Member Choose r => Eff (Exc CutFalse :> r) a -> Eff r a
      at src/Control/Eff/Cut.hs:69:9-63
    or from (Member Choose r1)
      bound by the type signature for
                 check :: Member Choose r1 =>
                          [Eff (Exc CutFalse :> r1) a1]
                          -> Union r1 (Eff (Exc CutFalse :> r1) a1) -> Eff r1 a1
      at src/Control/Eff/Cut.hs:(77,11)-(80,20)
      `r' is a rigid type variable bound by
          the type signature for
            call :: Member Choose r => Eff (Exc CutFalse :> r) a -> Eff r a
          at src/Control/Eff/Cut.hs:69:9
      `r1' is a rigid type variable bound by
           the type signature for
             check :: Member Choose r1 =>
                      [Eff (Exc CutFalse :> r1) a1]
                      -> Union r1 (Eff (Exc CutFalse :> r1) a1) -> Eff r1 a1
           at src/Control/Eff/Cut.hs:77:11
    Expected type: Eff r1 a1
      Actual type: Eff r a
    Relevant bindings include
      k :: a2 -> Eff (Exc CutFalse :> r1) a1
        (bound at src/Control/Eff/Cut.hs:82:32)
      u :: Union r1 (Eff (Exc CutFalse :> r1) a1)
        (bound at src/Control/Eff/Cut.hs:82:11)
      jq :: [Eff (Exc CutFalse :> r1) a1]
        (bound at src/Control/Eff/Cut.hs:82:8)
      check :: [Eff (Exc CutFalse :> r1) a1]
               -> Union r1 (Eff (Exc CutFalse :> r1) a1) -> Eff r1 a1
        (bound at src/Control/Eff/Cut.hs:81:2)
      loop :: [Eff (Exc CutFalse :> r) a]
              -> Free (Union (Exc CutFalse :> r)) a -> Eff r a
        (bound at src/Control/Eff/Cut.hs:71:2)
      call :: Eff (Exc CutFalse :> r) a -> Eff r a
        (bound at src/Control/Eff/Cut.hs:70:1)
    In the expression: loop jq (k x)
    In an equation for `check':
        check jq u | Just (Choose [x] k) <- prj u = loop jq (k x)
```

There are 3 more similar errors, showing that ghc can't infer that `r` and `a` are the same in 
`loop`, `next` and `check`.

The dirty and fast solution is to provide information that `r` and `a` are the same type in the scope of `call`
`ScopedTypeVariables` and `forall r a` do exactly that.